### DISCUSSION TITLE
How do I throw the bucket uploaded image URL to my table field "img_url"?

### DISCUSSION THREAD

**Discussion by rcapdepaula** [supabase/supabase#5196] [Questions] [answered]
I created a field in my table called "img_url". Can I use "add foreign key relation" to get the image URL from my bucket, or do I have to use another approach?

The image upload is inside the same form that contains the other fields.

**Comment by GaryAustin1** [comment, marked as answer]
Depending on what you are trying to do, not sure you need a foreign key relation.

If the file IS NOT in a "public" bucket then you really only store the path (bucket/dir/filename) and call createSignedUrl(dir/filename) with the bucket before first '/' split out, when you use it as it is time limited.  You already know the path if you have the filename so can just store it with your other fields. Obviously if all files for the table are in a fixed bucket, then don't store the bucket part.

If your file IS in a "public" bucket then you technically already know the url.  Calling getPublicUrl(path) just builds up the url from constants, it does not make an API call.  So it is fast.  Or you could build it yourself as it is your SupabaseUrl + '/storage/v1/object/public' + '/' +bucket + '/' + dir + '/' + filename.  So you can store that directly to your table, or just store path and build it later which ever works for you situation.

public url looks like: `https://madeupurlzbcdlxrq.supabase.in/storage/v1/object/public/bucket/optdir/filename.ext`  where everything up to bucket is fixed for your Supabase instance.

**Reply by rcapdepaula** [reply]
I do this and I get the url

`const { publicURL } = supabase.storage.from('images').getPublicUrl(`myfolder/${image.name}`)`

Now I have to figure out how to put it in my table field "img_url"

**Reply by GaryAustin1** [reply]
You either insert that column with the other data for your table if you have not inserted it yet, or update that row with the url with a call.  As I said you know the public url even before you upload it if you know the filename.

**Reply by rcapdepaula** [reply]
No, I put it in the 

```
const handleFilesUpload = async (e) => {
		const image = e.target.files[0]
		console.log(image)
		const { data, error } = await supabase.storage
			.from('images')
			.upload(`public/${image.name}`, image, {
				cacheControl: '3600',
				upsert: false
			})

		const { publicURL } = supabase.storage.from('images').getPublicUrl(`public/${image.name}`)

		console.log(publicURL)
	}
```

**Reply by GaryAustin1** [reply]
So if you have already inserted the rest of that row in your table, then you add a supabase update of that row after you get publicURL.   I don't know when you are inserting the rest of your form in relation to finishing the file upload.
Edit:  You know the publicURL BEFORE you make the .upload call actually if that helps.

**Reply by rcapdepaula** [reply]
Good idea

**Reply by rcapdepaula** [reply]
Works perfectly!!

	```const handleFilesUpload = async (e) => {
		const image = e.target.files[0]
		console.log(image)
		const { data, error } = await supabase.storage
			.from('images')
			.upload(`public/${image.name}`, image, {
				cacheControl: '3600',
				upsert: false
			})

		const { publicURL } = supabase.storage.from('images').getPublicUrl(`public/${image.name}`)

		 (formData = { ...formData, image_url: publicURL })
	}```

**Reply by rcapdepaula** [reply]
Thank you for your inspiration!

**Reply by GaryAustin1** [reply]
Just remember in a real environment with multiple users you will need to add a uid (or some identifier) to the path and even then have to decide what to do if the same filename is uploaded.  .upload upsert false will fail with the same name, or you need make the code replace if the same, or handle error and change the name slightly (like add a _2).  The last one would be a change to the URL and you have to make sure that gets dealt with.

**Reply by rcapdepaula** [reply]
Thanks!
